src/System.Runtime.Extensions/src/System/IO/PathHelper.Windows.cs

```diff
@@ -160,78 +158,34 @@ internal string Normalize(string path, bool fullCheck, bool expandShortPaths)
             }
             finally
             {
-                if (fullPath != null && fullPath.Capacity <= MaxBuilderSize)
-                {
-                    // Stash for reuse
-                    t_cachedOutputBuffer = fullPath;
-                }
+                // Clear the buffer and stash it
+                fullPath.Free();
+                s_fullPathBuffer = fullPath;
```

Comment:
The beginning of the method does:
```C#
StringBuffer fullPath = s_fullPathBuffer ?? new StringBuffer(PathInternal.MaxShortPath);
```
and then here at the end:
```C#
s_fullPathBuffer = fullPath;
```
That's an unnecessary write to a thread static in the case where we got it from the thread static at the beginning.  We should either:
1) null out the thread static at the beginning after getting the buffer from it, or
2) set the thread static when we allocate the new buffer and skip setting it here, e.g. remove this line and change the beginning to:
```C#
StringBuffer fullPath = t_fullPathBuffer ?? (t_fullPathBuffer = new StringBuffer(PathInternal.MaxShortPath));
```
We would do (1) if we're concerned about caching an instance that's too big (as was done in the StringBuilder case.  Otherwise we would do (2) and save on the TLS access.